packages/server/modules/multiregion/utils/dbSelector.ts

```diff
@@ -325,10 +342,23 @@ const setUpProjectReplication = async ({
         'Could not create subscription {subName} to {pubName} when setting up project replication for region {regionName}',
         to.public,
         {
-          cause: ensureError(err, 'Unknown database error when creating subscription'),
+          cause: ensureError(
+            sanitizeError(err),
+            'Unknown database error when creating subscription'
+          ),
           info: { subName, pubName, regionName }
         }
       )
-    if (!err.message.includes('already exists')) throw err
+    if (
+      !err.message.includes('already exists') &&
+      !err.message.includes('duplicate key value violates unique constraint')
+    )
+      throw sanitizeError(err)
   }
 }
+
+const sanitizeError = (err: unknown): unknown => {
+  if (!err) return err
+  if (get(err, 'where').includes('password='))
+    return { ...err, where: '[REDACTED AS IT CONTAINS CONNECTION STRING]' }
```

Comment:
is this not gonna break the Error subclassed type into a plain JS object? may not be an issue, but if some code relies on these actually being instances of `Error`, that kinda code may break.

ideally you'd re-throw a new Error here with the kind of message & info payload you want